Minimal changes to resolve #38

Liamsi opened the pull request:
Change from RIPEMD160 to SHA256
resolves #38 

If #53 is merged we can additionally remove ripemd160 from the glide file (or dep file if #52 get's merged first)

ebuchman commented:
Maybe it would be better if we have a `sha256truncated` utility function that we can just call with the `buf`, instead of having to create and write to the hasher and then slice it every time. At least that way if we change it again we just change it in one place

Liamsi commented:
Good point. It's a bit messy with all the `[:20]` right. My intention was to submit this PR as is (with the least changes possible) and further improve this in a followup PR shortly after (see https://github.com/tendermint/iavl/issues/38#issuecomment-386290979 and https://github.com/tendermint/iavl/issues/11). Will do as you suggested and submit another PR later.

Liamsi commented:
> Maybe it would be better if we have a sha256truncated utility function that we can just call with the buf, instead of having to create and write to the hasher and then slice it every time.

I took a slightly different approach and added a tiny package for the hasher. It makes the hasher a bit more visible in generated godoc and adds a bit more flexibility if we switched to another hasher. 

PTAL @ebuchman.

melekes commented on sha256truncated/sha256truncated.go:
```diff
@@ -0,0 +1,44 @@
+}
+
+func (d *digest) Sum(in []byte) []byte {
+	return d.Hash.Sum(in)[:20]
```
Size

Liamsi replied:
Done. Thanks!

melekes commented on sha256truncated/sha256truncated_test.go:
```diff
@@ -0,0 +1,32 @@
+
+func TestSize(t *testing.T) {
+	got := New().Size()
+	want := 20
```
Size
